### ISSUE TITLE
New metric: loss function

### ISSUE THREAD

**Issue by bnord** [deepskies/DeepDiagnostics#138]
Loss function as function of epoch (or batch)

1. top panel: Loss and validation data
2. bottom panel: residual between loss and validation curve; include dashed zero line

**Comment by bnord** [comment]
We'll probably need to specify the form of how the neural network (or whatever model) will export the loss data to be used by your module in deep diagnostics.

**Comment by bnord** [comment]
Hi @rzhaolx!

**Comment by rzhaolx** [comment]
Hi Everyone! Thank you for looping me in on this :)

If I can offer my two cents (I also don't know the aim of DeepDiagnostics very well, so apologies if it's not sensible!), the loss history reader/visualizer will benefit from being very "laid back," in the sense that we could simply ask the user to supply the training and validation loss history, as simple as numpy arrays or dictionaries? 

The reason for that suggestion is that, in my experience, there really isn't a uniform way to save the loss history, so only providing a pickle format can make the pipeline less flexible. The `inference` object in sbi does it in a dictionary, but if the user's own training loop (say with batch training or decreasing learning rate), then the user has to save their own loss.  

When using the `inference` object, I simply pickled the object, and when implementing my own training loop, I had to save them in dictionaries at the end of each epoch. I can happily supply code for this, or any other diagnostics that can be helpful :)
